plugins/mcreator-core/blockly/js/extensions_procedure_deps.js

```diff
@@ -0,0 +1,225 @@
+Blockly.Extensions.register('procedure_dependencies_tooltip',
+    function () {
+        let thisBlock = this;
+        this.setTooltip(function () {
+            const depList = javabridge.getDependencies(thisBlock.getFieldValue('procedure'));
+            if (depList.length === 0)
+                return javabridge.t('blockly.extension.procedure_dep_tooltip.empty');
+            let tooltip = javabridge.t('blockly.extension.procedure_dep_tooltip');
+            for (const dependency of depList)
+                tooltip += '\n' + dependency;
+            return tooltip;
+        });
+    });
+
+Blockly.Extensions.register('procedure_dependencies_onchange_mixin',
+    function () {
+        this.setOnChange(function (changeEvent) {
+            // Trigger the change only if a block is changed, moved, deleted or created
+            if ((changeEvent.type !== Blockly.Events.BLOCK_CHANGE ||
+                changeEvent.element !== 'field') &&
+                changeEvent.type !== Blockly.Events.BLOCK_CREATE &&
+                changeEvent.type !== Blockly.Events.BLOCK_MOVE) {
+                return;
+            }
+            const group = Blockly.Events.getGroup();
+            // Makes it so the block change and the unplug event get undone together.
+            Blockly.Events.setGroup(changeEvent.group);
+            const procedure = this.getFieldValue('procedure');
+            for (let i = 0; this.getField('name' + i); i++) {
+                const prevType = this.getInput('arg' + i).connection.getCheck();
+                let depType = null;
+                const depList = javabridge.getDependencies(procedure);

```

Comment:
Does it make sense to query dependencies every for-loop iteration? Looks like procedure is const so it can't really change here